tiled/client/sync.py

```diff
@@ -0,0 +1,112 @@
+import collections.abc
+import itertools
+
+from ..structures.core import StructureFamily
+from ..structures.data_source import DataSource, Management
+from .base import BaseClient
+
+
+def sync(
```

Comment:
It might be just me, but when I think about synchronizing two objects, I usually imagine them as equivalent (in terms of source and destination), and whichever object has older data gets updated. I wonder if it would be better to call this function something like "copy" or "transfer"?
Another option could be to define an `.update()` method (abstract method?) on the `BaseClient` itself and subclass it in specific clients; this way, we wouldn't need `_DISPATCH`. It could work as `update` in dictionaries but also recursively descend into catalogs, and this would make it unambiguous, which argument is the source and which which one is the destination.

Reply:
Renamed to `copy`